trl/trainer/grpo_trainer.py

```diff
@@ -967,7 +967,11 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
             mean_kl = (per_token_kl * completion_mask).sum() / completion_mask.sum()
             self._metrics[mode]["kl"].append(self.accelerator.gather_for_metrics(mean_kl).mean().item())
 
-        is_clipped = (coef_1 < (1 - self.epsilon_low)) | (coef_1 > (1 + self.epsilon_high))
+        # Compute the effective clip_radio
+        is_clipped = (
+            ((coef_1 < 1 - self.epsilon_low) & (advantages.unsqueeze(1) < 0)) |
+            ((coef_1 > 1 + self.epsilon_high) & (advantages.unsqueeze(1) > 0))
+        ).float()
```

Comment:
In the end, it's equivalent to this, right?

```suggestion
        # Compute the effective clip_ratio
        is_clipped = per_token_loss == -per_token_loss2
```